RecordHub.BasketService/RecordHub.BasketService.Infrastructure/Services/BasketService.cs

```diff
@@ -0,0 +1,156 @@
+﻿using AutoMapper;
+using FluentValidation;
+using MassTransit;
+using RecordHub.BasketService.Applicatation.Exceptions;
+using RecordHub.BasketService.Applicatation.Services;
+using RecordHub.BasketService.Domain.Entities;
+using RecordHub.BasketService.Domain.Models;
+using RecordHub.BasketService.Infrastructure.Data.Repositories;
+using RecordHub.Shared.Exceptions;
+using RecordHub.Shared.MassTransit.Models.Order;
+
+namespace RecordHub.BasketService.Infrastructure.Services
+{
+    public class BasketService : IBasketService
+    {
+        private readonly IBasketRepository _repo;
+        private readonly IPublishEndpoint _publishEndpoint;
+        private readonly IValidator<ShoppingCartItem> _validatorItem;
+        private readonly IValidator<ShoppingCart> _validatorCart;
+        private readonly IMapper _mapper;
+        private readonly CatalogChecker.CatalogCheckerClient _catalogCheckerClient;
+        public BasketService(IBasketRepository repo,
+            IValidator<ShoppingCartItem> validatorItem,
+            IValidator<ShoppingCart> validatorCart,
+            CatalogChecker.CatalogCheckerClient catalogCheckerClient,
+            IMapper mapper,
+            IPublishEndpoint publishEndpoint)
+        {
+            _repo = repo;
+            _publishEndpoint = publishEndpoint;
+            _validatorItem = validatorItem;
+            _validatorCart = validatorCart;
+            _catalogCheckerClient = catalogCheckerClient;
+            _mapper = mapper;
+        }
+
+        public Task<bool> ClearBasketAsync(string? userName, CancellationToken cancellationToken = default)
+        {
+            return _repo.ClearBasketAsync(userName);
+        }
+
+
+        public async Task<ShoppingCart?> GetBasketAsync(string? userId, CancellationToken cancellationToken = default)
+        {
+            return await _repo.GetBasketAsync(userId);
+        }
+
+        public async Task RemoveCartItemAsync(string? userName, string productId, CancellationToken cancellationToken = default)
+        {
+            var basket = await _repo.GetBasketAsync(userName);
+
+            if (basket == null)
+            {
+                throw new BasketIsEmptyException();
+            }
+
+            var itemToDelete = basket.Items.Find(i => i.ProductId.Equals(productId));
+            if (itemToDelete == null)
```

Comment:
here you don't have space before if statement, but have before prev if statement. Maybe there are some cases, where it's ok, but here are the same checks